cmd/release-controller/http.go

```diff
@@ -810,6 +810,57 @@ func (c *Controller) httpReleaseInfo(w http.ResponseWriter, req *http.Request) {
 		fmt.Fprintf(w, `</ul>`)
 	}
 
+	type releaseInfoStub struct {
+		Metadata   *struct {
+			Version  string   `json:"version"`
+			Previous []string `json:"previous"`
+		}               `json:"metadata"`
+	}
+
+	tagReleaseJson, err := c.releaseInfo.ReleaseInfo(tagPull)
+	var requiredUpgrades []string
+	var tagReleaseInfo releaseInfoStub
+	err = json.Unmarshal([]byte(tagReleaseJson), &tagReleaseInfo)
+	if err == nil && tagReleaseInfo.Metadata != nil {
+		requiredUpgrades = tagReleaseInfo.Metadata.Previous
+	}
+	if len(requiredUpgrades) > 0 {
+		sort.Slice(requiredUpgrades, func(i, j int) bool {
+			vi, erri := semverParseTolerant(requiredUpgrades[i])
+			vj, errj := semverParseTolerant(requiredUpgrades[j])
+			if erri == nil && errj == nil {
+				return vi.GT(vj)
+			}
+			return requiredUpgrades[i] > requiredUpgrades[j]
+		})
+		upgradesTo := c.graph.UpgradesTo(tag)
+		sort.Sort(newNewestSemVerFromSummaries(upgradesTo))
+		var releaseUpgradeScript string
+		if len(upgradesTo) == 0 {
+			releaseUpgradeScript = `document.getElementById("tests").insertAdjacentHTML("afterend", "<p id=\"upgrades-from\">Upgrades from:</p><ul></ul>");`
+		}
+		releaseUpgradeScript = fmt.Sprintf("%s\nvar upgrades_from = document.getElementById(\"upgrades-from\").nextSibling;",releaseUpgradeScript)
```

Comment:
This is very sketchy, why do we need a script?

Reply:
I don't want to add JS that we don't need.

Reply:
I would expect this to be rendered inline.

Reply:
I.e. as HTML, not as JS

Reply:
ok, was considering putting that in to make it doable with a goroutine, in case the release info was too consuming